reader.go

```diff
@@ -48,6 +48,17 @@ func (p *MasterPlaylist) DecodeFrom(reader io.Reader, strict bool) error {
 	return p.decode(buf, strict)
 }
 
+func (p *MasterPlaylist) WithCustomDecoders(customDecoders []CustomDecoder) Playlist {
```

Comment:
Please add comments to the public methods, which will help us to generate document correctly. @mjneil